### PR TITLE
feat(pipeline-template): Adds v2 plan command.

### PR DESCRIPTION
**Pull request by jtk54**

### CODE REVIEW

File: cmd/pipeline-template/plan.go [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+		return err
+	}
+
+	configJson, err := util.ParseJsonFromFileOrStdin(options.configPath)
+	if err != nil {
+		return err
+	}
+
+	planRequest := map[string]interface{}{
+    "schema": "v2",
```
Comment (ezimanyi) [inline cmd/pipeline-template/plan.go]: Spacing looks off here; in general looks like this file needs a `go fmt`

File: cmd/pipeline-template/plan.go [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+	"github.com/spinnaker/spin/util"
+	"net/http"
+)
+
+type PlanOptions struct {
+	*pipelineTemplateOptions
+	configPath        string
+}
+
+var (
```
Comment (ezimanyi) [inline cmd/pipeline-template/plan.go]: I think this can be `const`

File: cmd/pipeline-template/plan_test.go [hunk 3]
```diff
@@ -0,0 +1,251 @@
...
+	currentCmd := NewPlanCmd(pipelineTemplateOptions{})
+	rootCmd := getRootCmdForTest()
+	pipelineTemplateCmd := NewPipelineTemplateCmd(os.Stdout)
+	pipelineTemplateCmd.AddCommand(currentCmd)
+	rootCmd.AddCommand(pipelineTemplateCmd)
+
+	rootCmd.SetArgs(args)
+	err := rootCmd.Execute()
+	if err == nil {
+		t.Fatalf("Command failed with: %s", err)
```
Comment (ezimanyi) [inline cmd/pipeline-template/plan_test.go]: Might be more clear to make the message state we expected an exception but didn't get one

File: cmd/pipeline-template/plan_test.go [hunk 4]
```diff
@@ -0,0 +1,251 @@
...
+	args := []string{"pipeline-template", "plan", "--gate-endpoint", ts.URL} // Missing pipeline config file and stdin.
+	currentCmd := NewPlanCmd(pipelineTemplateOptions{})
+	rootCmd := getRootCmdForTest()
+	pipelineTemplateCmd := NewPipelineTemplateCmd(os.Stdout)
+	pipelineTemplateCmd.AddCommand(currentCmd)
+	rootCmd.AddCommand(pipelineTemplateCmd)
+
+	rootCmd.SetArgs(args)
+	err := rootCmd.Execute()
+	if err == nil {
```
Comment (ezimanyi) [inline cmd/pipeline-template/plan_test.go]: Should we check the error is as expected (ie, missing flag), or at least change the error message in the line below

File: cmd/pipeline-template/plan.go (hunk 1 above)
Comment (jtk54) [reply cmd/pipeline-template/plan.go]: Good catch.

File: cmd/pipeline-template/plan_test.go (hunk 4 above)
Comment (jtk54) [reply cmd/pipeline-template/plan_test.go]: Done.

File: cmd/pipeline-template/plan_test.go (hunk 3 above)
Comment (jtk54) [reply cmd/pipeline-template/plan_test.go]: Done.

File: cmd/pipeline-template/plan.go (hunk 2 above)
Comment (jtk54) [reply cmd/pipeline-template/plan.go]: Done.

File: cmd/pipeline-template/plan_test.go [hunk 5]
```diff
@@ -0,0 +1,251 @@
...
+   "attributeConstraints": {},
+   "enabled": true,
+   "payloadConstraints": {},
+   "pubsubSystem": "google",
+   "source": "jack",
+   "subscription": "super-derp",
+   "subscriptionName": "super-derp",
+   "type": "pubsub"
+  }
+ ],
```
Comment (lwander) [inline cmd/pipeline-template/plan_test.go]: The json tabbing between the test data is different -- intentional?

File: cmd/pipeline-template/plan_test.go [hunk 6]
```diff
@@ -0,0 +1,251 @@
...
+		} else {
+			w.WriteHeader(http.StatusNotFound)
+		}
+	}))
+	return httptest.NewServer(mux)
+}
+
+const testPipelineTemplatePlanResp = `
+{
+ "application": "app",
```
Comment (lwander) [inline cmd/pipeline-template/plan_test.go]: Is this guaranteed to be alphabetical when rendered by spin?

File: cmd/pipeline-template/plan_test.go (hunk 5 above)
Comment (jtk54) [reply cmd/pipeline-template/plan_test.go]: I passed one through `json_pp` and not the other, make them consistent.

File: cmd/pipeline-template/plan_test.go (hunk 6 above)
Comment (jtk54) [reply cmd/pipeline-template/plan_test.go]: Yep.

**Pull request merged**